meetings/2023-02-15.md

```diff
@@ -45,6 +51,28 @@
 
 ### Coordinating further work on authentication and authorization
 * SC: Proposed by eP. See also https://github.com/solid/authorization-panel/issues/325#issuecomment-1427113300
+* eP: Example scenario: https://github.com/solid/notifications/issues/134
+* eP: Last week Matthieu was here and hopefully Aaron will join tomorrow, Wouter will also be interested in this. Notification Panel currently has more clear use cases for AuthZ/AuthN scenarios acting as a client for notification sender and some notification receivers. I now have more experience with OIDC on the server side. We use Inrupt Solid-OIDC client libraries on the server and it's quite problematic. Two main problems: make sure we have a good AuthN approach for server-side client,Solid-OIDC doesn't address properly, and second clarify where AuthN ends and AuthZ starts. We made a mistake trying to separate clearly and have two separate panels because they are heavily intertwined. For example, if we look at Solid-OIDC, we have an OpenID provider that issues an ID token, latest Solid-OIDC doesn't get into access tokens... we use this ID token to get information about end user and the client but for browser applications it works pretty well, it's an app that runs on the device so it cannot securely store secrets. This flow is designed for web apps where user authenticates with OpenID provider which is responsible for issuing tokens and signing them. For server-side clients we don't have the same scenario because they can keep secrets. What Jackson originally proposed is a simpler mechanisms, he added Webhook subscription type. That could be a more general mechanism to pout in other places. I see this part is heavily missing, as Solid-OIDC will not fit for server-side clients. We are relying on OIDC ID tokens tokens... will not work with Inrupt libraries because of problems with refresh tokens and token rotation. So we need a different mechanism for that. From Gitter discussions I see Aaron agrees. We have a strong use cases to have a different approach to AuthN for server-side clients. Re: AuthN/AuthZ being too separated: in practice for server-side clients they should be able to authenticated independent of end user and may be not used by a person directly on the web browser, it can all be automated and the client is not used by a user per se, but on behalf of a user. For that case it'd be convenient that it can authenticate independently of user. And we still need to have a way for the client to say it acts on behalf of a social agent. This part is the AuthZ. I don't think we should strictly focus on AuthN because in that case trying to separate is counterproductive. 
```

Comment:
```suggestion
* eP: Last week Matthieu was here, and hopefully Aaron will join tomorrow; Wouter will also be interested in this. Notification Panel currently has more clear use cases for AuthZ/AuthN scenarios acting as a client for notification sender and some notification receivers. I now have more experience with OIDC on the server side. We use Inrupt Solid-OIDC client libraries on the server and they're quite problematic. Two main problems: first, make sure we have a good AuthN approach for server-side client, Solid-OIDC doesn't address properly; and second, clarify where AuthN ends and AuthZ starts. We made a mistake trying to have two separate panels because they are heavily intertwined. For example, if we look at Solid-OIDC, we have an OpenID provider that issues an ID token; latest Solid-OIDC doesn't get into access tokens... We use this ID token to get information about end user and the client, but for browser applications it works pretty well, it's an app that runs on the device so it cannot securely store secrets. This flow is designed for web apps where user authenticates with an OpenID provider which is responsible for issuing tokens and signing them. For server-side clients, we don't have the same scenario, because they can keep secrets. What Jackson originally proposed is a simpler mechanisms; he added Webhook subscription type. That could be a more general mechanism to put in other places. I see this part is heavily missing, as Solid-OIDC will not fit for server-side clients. We are relying on OIDC ID tokens... will not work with Inrupt libraries because of problems with refresh tokens and token rotation. So we need a different mechanism for that. From Gitter discussions, I see Aaron agrees. We have strong use cases to have a different approach to AuthN for server-side clients. Re: AuthN/AuthZ being too separated: in practice for server-side clients, they should be able to authenticate independent of end user, and may not be used by a person directly on the web browser; it can all be automated, and the client is not used by a user _per se_, but on behalf of a user. For that case, it'd be convenient that it can authenticate independently of a user. And we still need to have a way for the client to say it acts on behalf of a social agent. This part is the AuthZ. I don't think we should strictly focus on AuthN, because in that case, trying to separate is counterproductive. 
```